velox/common/encode/Base64.h

```diff
@@ -32,125 +29,125 @@ class Base64 {
-  static size_t calculateEncodedSize(size_t size, bool withPadding = true);
-
-  /// Encodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
+  static std::string encode(const folly::IOBuf* inputBuffer);
+  /// Encodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer'. The output must have enough space as
   /// returned by the calculateEncodedSize().
-  static void encode(const char* data, size_t size, char* output);
-
-  /// Decodes the specified encoded text.
-  static std::string decode(folly::StringPiece encoded);
-
-  /// Returns the actual size of the decoded data. Will also remove the padding
-  /// length from the input data 'size'.
-  static size_t calculateDecodedSize(const char* data, size_t& size);
-
-  /// Decodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
-  /// returned by the calculateDecodedSize().
-  static void decode(const char* data, size_t size, char* output);
-
-  static void decode(
-      const std::pair<const char*, int32_t>& payload,
-      std::string& output);
+  static void encode(const char* input, size_t inputSize, char* outputBuffer);
 
-  /// Encodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output' using URL encoding. The output must have enough
-  /// space as returned by the calculateEncodedSize().
-  static void encodeUrl(const char* data, size_t size, char* output);
-
-  /// Encodes the specified number of characters from the 'data' using URL
+  /// Encodes the specified number of characters from the 'input' using URL
   /// encoding.
-  static std::string encodeUrl(const char* data, size_t len);
-
-  /// Encodes the specified IOBuf data using URL encoding.
-  static std::string encodeUrl(const folly::IOBuf* data);
-
+  static std::string encodeUrl(const char* input, size_t inputSize);
   /// Encodes the specified text using URL encoding.
   static std::string encodeUrl(folly::StringPiece text);
+  /// Encodes the specified IOBuf data using URL encoding.
+  static std::string encodeUrl(const folly::IOBuf* inputBuffer);
+  /// Encodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer' using URL encoding. The output must have
+  /// enough space as returned by the calculateEncodedSize().
+  static void
+  encodeUrl(const char* input, size_t inputSize, char* outputBuffer);
 
+  /// Decodes the input Base64 encoded string.
+  static std::string decode(folly::StringPiece encodedText);
+  /// Decodes the specified encoded payload and writes the result to the
+  /// 'output'.
+  static void decode(
+      const std::pair<const char*, int32_t>& payload,
+      std::string& output);
+  /// Decodes the specified number of characters from the 'data' and writes the
```

Comment:
`data` -> `input`, `output` -> `outputBuffer`.

Reply:
Updated the code